query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/discover/LdapTestAttributeMappings.java

```diff
@@ -0,0 +1,161 @@
+        conn = new LdapConnectionField().useDefaultRequired();
+        bindInfo = new LdapBindUserInfo().useDefaultRequired();
+        settings = new LdapSettingsField().useDefaultUserAttributes();
+        updateArgumentPaths();
+
+        utils = new LdapTestingUtils();
+    }
+
+    @Override
+    public List<DataType> getArguments() {
+        return ImmutableList.of(conn, bindInfo, settings);
+    }
+
+    @Override
+    public FunctionField<BooleanField> newInstance() {
+        return new LdapTestAttributeMappings();
+    }
+
+    @Override
+    public BooleanField performFunction() {
+        try (LdapConnectionAttempt connectionAttempt = utils.bindUserToLdapConnection(conn,
+                bindInfo)) {
+            addMessages(connectionAttempt);
+
+            if (containsErrorMsgs()) {
+                return new BooleanField(false);
+            }
+
+            Connection ldapConnection = connectionAttempt.result();
+
+            checkUserDirExists(ldapConnection);
+
+            // Short-circuit return here, if either the user or group directory does not exist
+            if (containsErrorMsgs()) {
+                return new BooleanField(false);
+            }
+
+            settings.attributeMapField()
+                    .getList()
+                    .stream()
+                    .map(ClaimsMapEntry::claimValueField)
+                    .filter(claim -> !mappingAttributeFound(ldapConnection, claim.getValue()))
+                    .forEach(claim -> addArgumentMessage(mappingAttributeNotFoundError(claim.path())));
+
+            if (containsErrorMsgs()) {
+                return new BooleanField(false);
+            }
+        } catch (IOException e) {
+            LOGGER.warn("Error closing LDAP connection", e);
+        }
+
+        return new BooleanField(!containsErrorMsgs());
+    }
+
+    /**
+     * Confirms that at least one user exists in the user base with the attribute provided
+     *
+     * @param ldapConnection connection used to query ldap
+     */
+    private void checkUserDirExists(Connection ldapConnection) {
+        LdapDistinguishedName dirDn = settings.baseUserDnField();
+        boolean dirExists = !utils.getLdapQueryResults(ldapConnection,
+                dirDn.getValue(),
+                Filter.present("objectClass")
+                        .toString(),
+                SearchScope.BASE_OBJECT,
+                1)
+                .isEmpty();
+
+        if (!dirExists) {
+            addArgumentMessage(dnDoesNotExistError(dirDn.path()));
+        }
+    }
+
+    /**
+     * Checks the baseUserDn for users.
+     *
+     * @param ldapConnection connection used to query ldap
+     * @param mapAttribute
+     */
+    private boolean mappingAttributeFound(Connection ldapConnection, String mapAttribute) {
+        return !utils.getLdapQueryResults(ldapConnection,
+                settings.baseUserDn(),
+                Filter.and(Filter.present(settings.usernameAttribute()),
```

Comment:
If we have an entry under the base user DN that has our claim-mapped attribute but isn't actually a user, it's like that user isn't there at all.